hazelcast/src/test/java/com/hazelcast/internal/nearcache/AbstractNearCacheBasicTest.java

```diff
@@ -259,139 +261,139 @@ private void whenPutAllIsUsed_thenNearCacheShouldBeInvalidated(boolean useNearCa
-        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(true, false);
+        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(true, DataStructureMethods.REPLACE);
     }
 
     /**
-     * Checks that the Near Cache values are eventually invalidated when {@link DataStructureAdapter#replace(Object, Object)}
-     * is used.
+     * Checks that the Near Cache is eventually invalidated when {@link DataStructureMethods#REPLACE} is used.
      *
      * This variant uses the {@link NearCacheTestContext#dataAdapter}, so we need to configure Near Cache invalidation.
      */
     @Test
     public void whenReplaceIsUsed_thenNearCacheShouldBeInvalidated_withUpdateOnDataAdapter() {
         nearCacheConfig.setInvalidateOnChange(true);
-        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(false, false);
+        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(false, DataStructureMethods.REPLACE);
     }
 
     /**
-     * Checks that the Near Cache values are eventually invalidated when
-     * {@link DataStructureAdapter#replace(Object, Object, Object)} is used.
+     * Checks that the Near Cache is eventually invalidated when {@link DataStructureMethods#REPLACE_WITH_OLD_VALUE} is used.
      *
      * This variant uses the {@link NearCacheTestContext#nearCacheAdapter}, so there is no Near Cache invalidation necessary.
      */
     @Test
     public void whenReplaceWithOldValueIsUsed_thenNearCacheShouldBeInvalidated_withUpdateOnNearCacheAdapter() {
-        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(true, true);
+        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(true, DataStructureMethods.REPLACE_WITH_OLD_VALUE);
     }
 
     /**
-     * Checks that the Near Cache values are eventually invalidated when
-     * {@link DataStructureAdapter#replace(Object, Object, Object)} is used.
+     * Checks that the Near Cache is eventually invalidated when {@link DataStructureMethods#REPLACE_WITH_OLD_VALUE} is used.
      *
      * This variant uses the {@link NearCacheTestContext#dataAdapter}, so we need to configure Near Cache invalidation.
      */
     @Test
     public void whenReplaceWithOldValueIsUsed_thenNearCacheShouldBeInvalidated_withUpdateOnDataAdapter() {
         nearCacheConfig.setInvalidateOnChange(true);
-        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(false, true);
+        whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(false, DataStructureMethods.REPLACE_WITH_OLD_VALUE);
     }
 
-    private void whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(boolean useNearCacheAdapter, boolean useOldValue) {
+    private void whenReplaceIsUsed_thenNearCacheShouldBeInvalidated(boolean useNearCacheAdapter, DataStructureMethods method) {
         NearCacheTestContext<Integer, String, NK, NV> context = createContext();
+        DataStructureAdapter<Integer, String> adapter = useNearCacheAdapter ? context.nearCacheAdapter : context.dataAdapter;
+        assumeThatMethodIsAvailable(adapter, method);
```

Comment:
This is all we need to do to check if the required method is available in the executed scenario. If the method is not available `assumeThat()` will stop the test without a failure.